helper/pointer/pointer.go

```diff
@@ -14,3 +18,21 @@ func Copy[A any](a *A) *A {
 	na := *a
 	return &na
 }
+
+// Primitive represents basic types that are safe to do basic comparisons by
+// pointer dereference (checking nullity first).
+type Primitive interface {
+	constraints.Ordered
+}
+
+// Compare returns whether a and b contain the same value.
+//
+// May only be used on pointers to primitive types, where the comparison is
+// guaranteed to be sensible. For complex types (i.e. structs) consider implementing
+// an Equals method.
+func Compare[P Primitive](a, b *P) bool {
```

Comment:
I always think of "Compare" as returning differences (like some sort of Diff struct) and "Equals" as a bool. No big deal though. Folks will know what's up either way I'm sure.

Reply:
The thing with "Equals" is it sounds the same as the built-in pointer equals `==`. Maybe `pointer.Eq` ? `pointer.Same` ?